caffeine@patapon.info/extension.js

```diff
@@ -850,21 +869,78 @@ class Caffeine extends QuickSettings.SystemIndicator {
                 if (this._state) {
                     this._handleToggleClick();
                 }
+
                 return GLib.SOURCE_REMOVE;
             });
         }
     }
 
-    _printTimer(seconds) {
-        const hours = Math.floor(seconds / 3600).toString().padStart(2, '0');
-        const min = Math.floor((seconds % 3600) / 60).toString().padStart(2, '0');
-        const sec = Math.floor(seconds % 60).toString().padStart(2, '0');
-        // Print Timer in system Indicator and Toggle menu subLabel
-        if (hours !== '00') {
-            this._updateLabelTimer(hours + ':' + min + ':' + sec);
-        } else {
-            this._updateLabelTimer(min + ':' + sec);
+    _toISO8601(rawSeconds) {
+        const hh = Math.floor(rawSeconds / 3600);
+        const mm = Math.floor((rawSeconds % 3600) / 60);
+        const ss = Math.floor(rawSeconds % 3600) % 60;
+
+        let iso8601 = `${String(mm).padStart(2, '0')}:${String(ss).padStart(2, '0')}`;
+        if (hh !== 0) {
+            iso8601 = `${String(hh).padStart(2, '0')}:${iso8601}`;
+        }
+        return iso8601;
+    }
+
+    _getSystemNotificationSource() {
+        if (!this._systemNotificationSource) {
+            const notificationPolicy = new MessageTray.NotificationGenericPolicy();
+
+            this._systemNotificationSource = new MessageTray.Source({
+                title: _('Caffeine'),
+                icon: this._iconActivated,
+                policy: notificationPolicy
+            });
+
+            this._systemNotificationSource.connect('destroy', (_source) => {
+                this._systemNotificationSource = null;
+            });
+
+            Main.messageTray.add(this._systemNotificationSource);
         }
+
+        return this._systemNotificationSource;
+    }
+
+    _sendSystemNotification() {
+        const systemSource = this._getSystemNotificationSource();
+
+        if (this._systemNotification) {
+            this._systemNotification.destroy();
+            this._systemNotification = null;
+        }
+
+        this._systemNotification = new MessageTray.Notification({
+            source: systemSource,
+            title: _('You are running out of Caffeine'),
+            body: _('select a duration if you still want to continue'),
```

Comment:
```suggestion
            body: _('Select a duration if you want to extend it'),
```

Missed a capital, I think "still want to continue" implies we're confirming an existing action